src/blockdata/script.rs

```diff
@@ -312,13 +315,40 @@ impl Script {
     }
 
     /// Generates P2WPKH-type of scriptPubkey
+    #[deprecated(since = "0.28.0", note = "use Script::new_v0_p2wpkh method instead")]
     pub fn new_v0_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash.to_vec())
+        Script::new_v0_p2wpkh(pubkey_hash)
+    }
+
+    /// Generates P2WPKH-type of scriptPubkey
+    pub fn new_v0_p2wpkh(pubkey_hash: &WPubkeyHash) -> Script {
+        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash[..])
     }
 
     /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
+    #[deprecated(since = "0.28.0", note = "use Script::new_v0_p2wsh method instead")]
     pub fn new_v0_wsh(script_hash: &WScriptHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &script_hash.to_vec())
+        Script::new_v0_p2wsh(script_hash)
+    }
+
+    /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
+    pub fn new_v0_p2wsh(script_hash: &WScriptHash) -> Script {
+        Script::new_witness_program(WitnessVersion::V0, &script_hash[..])
+    }
+
+    /// Generates P2TR for script spending path using an internal public key and some optional
+    /// script tree merkle root.
+    pub fn new_v1_p2tr<C: Verification>(secp: &Secp256k1<C>, internal_key: UntweakedPublicKey, merkle_root: Option<TapBranchHash>) -> Script {
+        let (output_key, _) = internal_key.tap_tweak(secp, merkle_root);
+        Script::new_witness_program(WitnessVersion::V1, &output_key.serialize())
+    }
+
+    /// Generates P2TR for key spending path for a known [`TweakedPublicKey`].
+    ///
+    /// NB: Make sure that the used key is indeed tweaked (for instance, it comes from `rawtr`
+    /// descriptor content); otherwise please use [`Script::new_v1_p2tr`] method.
```

Comment:
I think this comment here is not needed since tweaking is tracked in the type. Such comment is more appropriate for `assume_tweaked`.

Reply:
The commends are different between types, at least should be

Reply:
I mean you don't have a "make sure this is dereferencable" comment on `&T` documentation because the type already guarantees it and you have that comment on the "constructor".